app/domain/authentication/authn_k8s/copy_text_to_file_in_container.rb

```diff
@@ -42,29 +42,28 @@ def copy_text_to_file_in_container
       #
       # We redirect the output to a log file on the authn-client container
       # that will be written in its logs for supportability.
-      def set_file_content_script(path, content, mode)
+      def bash_script(path, content, mode)
         tmp_cert = "#{path}.tmp"
+        <<~BASH_SCRIPT
```

Comment:
We should use `mktemp` as a thread safe and standard way to make the temp file (avoiding possible collisions, eg)

Reply:
if i understand correctly, this command will create the temporary file in a temp directory instead of the current one. This is a security concern as we want the temp cert to be in the same directory as the regular one so it has the same permissions.

Reply:
I see.  Let's create a random filename then?

Reply:
Doesn't Ruby has the equivalent to [c#](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=netcore-3.1)?

Reply:
> Let's create a random filename then?
yeah, that's a good idea.